compiler/rustc_passes/src/dead.rs

```diff
@@ -1185,19 +1158,11 @@ fn check_mod_deathness(tcx: TyCtxt<'_>, module: LocalModDefId) {
         let def_kind = tcx.def_kind(item.owner_id);
 
         let mut dead_codes = Vec::new();
-        // if we have diagnosed the trait, do not diagnose unused methods
-        if matches!(def_kind, DefKind::Impl { .. })
+        // if we have diagnosed the trait, do not diagnose unused assoc items
+        if matches!(def_kind, DefKind::Impl { of_trait: false })
             || (def_kind == DefKind::Trait && live_symbols.contains(&item.owner_id.def_id))
```

Comment:
Could you reformulate the comment, or move it one line lower? It took me a while to understand the relationship between the "if we have diagnosed the trait" and the `of_trait: false`.